src/components/ModalWithForm/ModalWithForm.css

```diff
@@ -143,42 +148,57 @@
   opacity: 0.8;
 }
 
-@media screen and (max-width: 950px){
-  .modal__title{
+.modal__error {
+  left: 0;
+  right: 0;
+  margin-left: auto;
+  margin-right: auto;
+  width: 100%;
+  position: absolute;
+  top: -20px;
+  font-family: "Inter";
+  font-style: normal;
+  font-weight: 400;
+  font-size: 12px;
+  line-height: 15px;
+  color: #ff0000;
+  cursor: disabled;
+}
+
+@media screen and (max-width: 950px) {
+  .modal__title {
     margin-bottom: 22px;
   }
 }
 
-@media screen and (max-width: 550px){
-  .modal{
+@media screen and (max-width: 550px) {
+  .modal {
     align-items: flex-end;
   }
-  .modal__container{
+  .modal__container {
     display: block;
     width: 100%;
     position: relative;
     display: flex;
     flex-direction: column;
   }
 
-  .modal__form{
+  .modal__form {
     width: 100%;
     height: 512px;
     padding-left: 16px;
     padding-right: 16px;
   }
 
-  .modal__button-closed{
+  .modal__button-closed {
     width: 24px;
     height: 24px;
     position: absolute;
-    top:0;
+    top: 0;
     right: 16px;
   }
 
-  .modal__button{
+  .modal__button {
     margin-top: 23px;
   }
-
```

Comment:
* you have a lot of constant server requests. See here https://disk.yandex.ru/i/CmeAn8WkOyJf6Q  It's because you added `savedCards` for the `useEffect` with the token check. See here https://disk.yandex.ru/i/M4qfma3m3k9ivw . The dependencies array should be **empty** like this `}, []);`